feat: update nodejs bindings

laumair opened the pull request:
# Description of change

- Change address outputs from an array to a list (object);
- Add `id` to Message interface;
- Make `confirmed` property in `Message` interface non-optional;
- Add `reattachmentMessageId` to Message interface;
- Add `type: ‘Transaction’` to Transaction interface;
- Nest `Transaction.essence` into `Transaction.data`;
- Make `metadata` in UTXOInput interface non-optional;
- Change `is_spent` in UTXOInput interface to `isSpent`;
- Add `Indexation` payload to RegularEssence interface;
- Add unlock_blocks to Message interface — (TODO) Should be camelCase including public_key prop;
- Add `Milestone` payload interface

## Links to any relevant issues

Fixes https://github.com/iotaledger/wallet.rs/issues/871

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

Commit:
feat: update nodejs bindings

- Change address outputs from an array to a list (object);
- Add “id” to Message interface;
- Make “confirmed” property in Message interface non-optional;
- Add “reattachmentMessageId” to Message interface;
- Add “type: ‘Transaction’” to Transaction interface;
- Nest “Transaction.essense” into “Transaction.data”;
- Make “metadata” in UTXOInput interface non-optional;
- Change “is_spent” in UTXOInput interface to “isSpent”;
- Add “Indexation” payload to RegularEssence interface;
- Add unlock_blocks to Message interface — (TODO) Should be camelCase including public_key prop;
- Add Milestone payload interface

Thoralf-M commented on bindings/nodejs/lib/index.d.ts:
```diff
@@ -16,7 +17,13 @@ export declare enum MessageType {
   inputs: Input[];
   outputs: Output[];
-  payload?: Payload[];
+  payload: {
```
Should still be optional I think

Thoralf-M commented on bindings/nodejs/lib/index.d.ts:
```diff
@@ -30,12 +37,12 @@ export declare type Essence = {
 export declare interface UtxoInput {
   input: string;
-  metadata?: {
+  metadata: {
```
This here should also still be optional

Commit:
feat: make unlock_blocks camelCase

Commit:
fix (nodejs): address comments

- make metadata in UtxoInput interface optional;
- make payload in RegularEssence interface optional

Commit:
chore: add to .changes/ for next release
